DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java

```diff
@@ -527,68 +528,79 @@ private String paramsTypesToString(Class<?>[] paramTypes) {
-                    paramType =
-                            convertDBArray(paramType, parameterizedTypes[candidateParamTypes.length - 1] == null ? null
-                                    : parameterizedTypes[candidateParamTypes.length - 1][0]);
-                }
+            if (isDbArray(paramType) && candidateParamTypes[candidateParamTypes.length - 1].isArray()) {
+                paramType = convertDBArray(paramType, parameterizedTypes[candidateParamTypes.length - 1] == null ? null
+                        : parameterizedTypes[candidateParamTypes.length - 1][0]);
+            }
 
-                if (candidateParamTypes.length == paramTypes.length && paramType.isArray()) {
-                    if (!isAssignableFrom(candidateParamTypes[candidateParamTypes.length - 1], paramType)) {
-                        acceptable = false;
-                    }
-                } else {
-                    final Class<?> lastClass = candidateParamTypes[candidateParamTypes.length - 1].getComponentType();
+            if (candidateParamTypes.length == paramTypes.length && paramType.isArray()) {
+                if (!canAssignType(candidateParamTypes[candidateParamTypes.length - 1], paramType)) {
+                    return;
+                }
+            } else {
+                final Class<?> lastClass = candidateParamTypes[candidateParamTypes.length - 1].getComponentType();
 
-                    for (int i = candidateParamTypes.length - 1; i < paramTypes.length; i++) {
-                        paramType = paramTypes[i];
+                for (int i = candidateParamTypes.length - 1; i < paramTypes.length; i++) {
+                    paramType = paramTypes[i];
 
-                        if (isDbArray(paramType) && lastClass.isArray()) {
-                            paramType = convertDBArray(paramType,
-                                    parameterizedTypes[i] == null ? null : parameterizedTypes[i][0]);
-                        }
+                    if (isDbArray(paramType) && lastClass.isArray()) {
+                        paramType = convertDBArray(paramType,
+                                parameterizedTypes[i] == null ? null : parameterizedTypes[i][0]);
+                    }
 
-                        if (!isAssignableFrom(lastClass, paramType)) {
-                            acceptable = false;
-                            break;
-                        }
+                    if (!canAssignType(lastClass, paramType)) {
+                        return;
                     }
                 }
             }
+        }
 
-            if (acceptable) {
-                accepted.add(candidate);
-            }
+        accepted.add(candidate);
+    }
+
+    private static boolean canAssignType(final Class<?> candidateParamType, final Class<?> paramType) {
+        boolean canAssignWithoutUnbox = isAssignableFrom(candidateParamType, paramType);
+
+        if (paramType.isArray() && candidateParamType.isArray()
```

Comment:
This should only work for object arrays, unless or until we add implicit primitive array conversion.

Reply:
Captured the follow-up conversation in #1301. I added the biggest concern and the biggest benefit. As we discussed offline; it's not obvious if it's worth doing.